### ISSUE TITLE
Clarification on deserialization

### ISSUE THREAD

**Issue by ChrisGitWorkshop** [kbss-cvut/jb4jsonld#67]
Hi,

Can i check whether a nested object in which the child nodes does only have "@id" without "@type" will perform the deserialization using the jb4jsonld ?

This use case come when, i need to refer an already existing person object in the payload which only have a id.

I saw the config property `assumeTargetType = true` to deal with this (As per my understanding). But it doesn't work as I expected. I am getting the below error There are unresolved references to objects. (Though i noticed, i am getting the targetted Type in debug, but since its not PersistenceProperties#IDENTIFIER_TYPE, not added).

When i do a quick analysis,  i found below.

In ```javaCollectionDeserializer#resolveValue(Map)``` , there is a conditiona as below
```java
} else if (value.size() == 1 && value.containsKey(JsonLd.ID)) {
   handleReferenceNodeInCollection(value, targetType);
}
``` 
So from the above code, the execution goes all the way to `DefaultInstanceBuilder#addNodeForREference(String)`. From here it check for `canDirectlyAddNodeReference (targetType)`, where it check whther the targetted type is any of PersistenceProperties#IDENTIFIER_TYPE, since its adding to pendingReferenceRegistry. 

```java
/**
 * Supported identifier Java types.
 * <p>
 * This set represents the Java types that can be used as entity identifiers or as {@link
 * cz.cvut.kbss.jopa.model.annotations.OWLObjectProperty} property values.
 */
public static final Set<Class<?>> IDENTIFIER_TYPES = Set.of(URI.class, URL.class, String.class);
```

Example object given below.

```json
{
    "@context": {        
        "example": "https://example.model/data#"
    },
    "@type":"example:SocialMediaAccount",
    "example:profileName": "MyNameIsXYZ",
    "example:bestFriend": {
        "@id": "https://socialmedial/account/person/MyBestiee"
    }
}
```

Any input would be appreciated.

Thanks in advacne,
Chris.

**Comment by ChrisGitWorkshop** [comment]
Dear Team,

Many thanks for finding time to look into this issue, appriciate it.

The suggestion you have put forward sounds good to me. Becuase, when i had a quick look at the jsonld doc, it implicitly saying that `@type` is not a mandatory filed in the object (I am not an expert in this data structure though :) ). Hence this might be helpful if we could implement it (Much better if we have some configuration in place,  for instance as you mentioned to make use of `ASSUME_TARGET_TYPE` property, so that people can make a choice).


Regards,
Chris.

**Comment by ledsoft** [comment]
I created a follow-up ticket for the discussed enhancement - #69.

**Comment by ledsoft** [comment]
Implemented in 0.15.0.
